lib/inets/src/http_server/mod_dir.erl

```diff
@@ -211,7 +227,7 @@ percent_encode(URI) when is_binary(URI) ->
     << <<(uri_encode_binary(Char, Reserved))/binary>> || <<Char>> <= URI >>.
 
 reserved() ->
-    sets:from_list([$;, $:, $@, $&, $=, $+, $,, $/, $?,
+    sets:from_list([$;, $:, $@, $&, $=, $+, $,, $?, % $/,
```

Comment:
You're right. RequestURI is already (supposed to be) percent-encoded when we get it, so we'll skip percent-encoding it in `format/2`. (Request-URI is defined in https://www.ietf.org/rfc/rfc1945.html#section-5.1.2, i.e. it's the path in the first line of a HTTP/1 request, e.g. `GET /request%20uri HTTP/1.0`.)

Unfortunately, we can't test it because httpd shows 404 for URIs with percent-encoded chars in the path, before we get to mod_dir. (That's another bug.)

Reply:
- Builds failed due to dialyzer warning, parts of encoding for binary variant seems unused now (it should be reproducible by calling run-dialyzer script in local repo)
- When dialyzer is fixed, builds are fine and we get repeated test results - I would consider merging the work done so far in this PR. IMHO work assumed in GH-4855 is done.
- New GH issue for link+percent=>404 bug discovered could be a good idea (its description could include a task of verifying our assumption on RequestURI received in mod_dir - however if re-encoding would be needed I would consider checking if uri_string functionality could be re-used in mod_dir)
- What do you think?